virtual-setup/working-with-your-own-data.md

```diff
@@ -0,0 +1,233 @@
+For that, we can use the `-O`, or `output` option with our `wget` command and specify a file path.  
+
+*Template:*
+```
+wget -O <FILE_PATH_TO_SAVE_TO> <URL>
+```
+
+*Specific example:* Here's an example where we will download that same array express file, but instead save it to the `data` folder and call it `some_array_data.zip`.
+(Best to keep the file extension consistent to avoid troubles!)
+
+```
+wget -O data/some_array_data.zip https://www.ebi.ac.uk/arrayexpress/files/E-GEOD-67851/E-GEOD-67851.processed.1.zip
+```
+
+`-O` is one of many `wget` command options.
+To see the complete list of `wget` options, use the command: `wget -h` in bash.
+
+[See more `wget` examples](https://www.tecmint.com/10-wget-command-examples-in-linux/)
+
+### Upload data from your own computer
+
+If the data you want to use is stored locally on your computer, here's how we recommend uploading it to the RStudio Server.
+
+**Step 1)** We recommend you compress your data folder into a single zip file.
+
+For most operating systems, you can right-click on your data folder, and choose `Compress` to zip up your files
+- [Windows zipping](https://support.microsoft.com/en-us/help/14200/windows-compress-uncompress-zip-files)
+- [Mac zipping](https://www.imore.com/how-compress-file-your-mac)
+
+For reference, here's how you [compress files from the command line](https://coolestguidesontheplanet.com/how-to-compress-and-uncompress-files-and-folders-in-os-x-lion-10-7-using-terminal/).
+
+**Step 2)** Once your data is compressed to a single file, [navigate to your RStudio session](./rstudio-login.md).
+
+**Step 3)** Use the `Upload button` to choose your compressed data folder.
+
+This button is in the lower right panel of your RStudio session:
+![Upload button](screenshots/upload-button.png)
+
+A mini screen will pop up asking you to choose the file you want to upload:
+
+![Choose file](screenshots/upload-choose-file.png)
+
+Choose your compressed data file, and click `OK`.
+This may take some time, particularly if you have a large dataset.  
+
+When the server is finished uploading your data, you should see your file in your `home` directory!
+
+### Reading the data into your R environment.
+
+This step is very dependent on the format of your data and what you are planning to do with it!
+If your file is a TSV, CSV, or RDS file, follow the examples in the [`intro-to-R-tidyverse/intro-to-tidyverse.Rmd` notebook](intro-to-R-tidyverse/03-intro_to_tidyverse.Rmd).
+
+We do NOT recommend clicking on the file in the RStudio panel to load it into your R environment, this typically won't work for anything that's not a very small file.
+
+Plus, for reproducibility purposes, you should write the data reading step into your notebook analyses!
+
+## Downloading files
+
+Any files on the RStudio server you would like to save to your computer you can export.
+
+**Step 1)** We recommend you compress the files you want copied to your computer into a zip file.
```

Comment:
The server will automatically do this step (at least for folders or multiple files). They do not need to zip on their own.

Reply:
Oh! Good to know!